File: Java version/app/src/main/java/it/unical/mat/embasp/specializations/dlvhex/desktop/DLVHEXDesktopService.java
package it.unical.mat.embasp.specializations.dlvhex.desktop;

import it.unical.mat.embasp.base.Output;
import it.unical.mat.embasp.platforms.desktop.DesktopService;
import it.unical.mat.embasp.specializations.dlvhex.DLVHEXAnswerSets;

public class DLVHEXDesktopService extends DesktopService {

	public DLVHEXDesktopService(final String exe_path) {
		super(exe_path);
		
		load_from_STDIN_option = "--";
	}

	@Override
	protected Output getOutput(final String output, final String error) {
		return new DLVHEXAnswerSets(output, error);
	}
}

File: CSharp version/it/unical/mat/embasp/specializations/dlvhex/desktop/DLVHEXDesktopService.cs
namespace it.unical.mat.embasp.specializations.dlvhex.desktop
{
    using Output = it.unical.mat.embasp.@base.Output;
    using DesktopService = it.unical.mat.embasp.platforms.desktop.DesktopService;

    public class DLVHEXDesktopService : DesktopService
    {
        public DLVHEXDesktopService(string exe_path) : base(exe_path) => load_from_STDIN_option = "--";

        protected internal override Output GetOutput(string output, string error) => new DLVHEXAnswerSets(output, error);
    }
}